company-template.el

```diff
@@ -33,6 +33,7 @@
   (let ((keymap (make-sparse-keymap)))
     (define-key keymap [tab] 'company-template-forward-field)
     (define-key keymap (kbd "TAB") 'company-template-forward-field)
+    (define-key keymap (kbd "C-d") 'company-template-clear-field)
```

Comment:
When we're editing a variable name, the template field is gone, right?

A template field is an overlay. We should probably create a separate keymap for them, with just this one binding.